internal/security/config/command/proxy/oauth2/command.go

```diff
@@ -0,0 +1,120 @@
+//
+// Copyright (c) 2020 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0'
+//
+
+package oauth2
+
+import (
+	"bytes"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/edgex-go/internal/security/config/interfaces"
+	"github.com/edgexfoundry/edgex-go/internal/security/proxy/config"
+	"github.com/edgexfoundry/edgex-go/internal/security/secretstoreclient"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	"github.com/edgexfoundry/go-mod-core-contracts/clients/logger"
+)
+
+const (
+	CommandName string = "oauth2"
+)
+
+type cmd struct {
+	loggingClient logger.LoggingClient
+	client        internal.HttpCaller
+	configuration *config.ConfigurationStruct
+	clientID      string
+	clientSecret  string
+}
+
+func NewCommand(
+	lc logger.LoggingClient,
+	configuration *config.ConfigurationStruct,
+	args []string) (interfaces.Command, error) {
+
+	cmd := cmd{
+		loggingClient: lc,
+		client:        secretstoreclient.NewRequestor(lc).Insecure(),
+		configuration: configuration,
+	}
+	var dummy string
+
+	flagSet := flag.NewFlagSet(CommandName, flag.ContinueOnError)
+	flagSet.StringVar(&dummy, "confdir", "", "") // handled by bootstrap; duplicated here to prevent arg parsing errors
+
+	flagSet.StringVar(&cmd.clientID, "client_id", "", "Optional manually-specified OAuth2 client_id. Will be generated if not present. Equivalent to a username.")
+	flagSet.StringVar(&cmd.clientSecret, "client_secret", "", "Optional manually-specified OAuth2 client_secret. Will be generated if not present. Equivalent to a password.")
+
+	err := flagSet.Parse(args)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse command: %s: %w", strings.Join(args, " "), err)
+	}
+	if cmd.clientID == "" {
+		return nil, fmt.Errorf("%s proxy oauth2: argument --client_id is required", os.Args[0])
+	}
+	if cmd.clientSecret == "" {
+		return nil, fmt.Errorf("%s proxy oauth2: argument --client_secret is required", os.Args[0])
```

Comment:
Isn't this optional?

Reply:
client_id and client_secret are required to be passed to the OAuth2 token creation endpoint.

The new utility separates creation of the user from creating the token.  The old utility did both actions in one step, which is broken.